client/llb/resolver.go

```diff
@@ -34,8 +34,16 @@ type ImageMetaResolver interface {
 	ResolveImageConfig(ctx context.Context, ref string, opt ResolveImageConfigOpt) (digest.Digest, []byte, error)
 }
 
+type ContainerImageResolverType string
+
+const (
+	ResolverTypeRegistry  ContainerImageResolverType = "registry"
+	ResolverTypeOCILayout ContainerImageResolverType = "oci-layout"
+)
+
 type ResolveImageConfigOpt struct {
 	Platform    *ocispecs.Platform
 	ResolveMode string
 	LogName     string
+	Source      ContainerImageResolverType // default is ResolverTypeRegistry
```

Comment:
This should be called `ResolverType`

Reply:
> This should be called ResolverType

Fixed across the board, both client-side and daemon-side.

Reply:
> Maybe int type with iota so we don't have this special case where "" == "registry" needs to be considered true.

Changed to `int` with type `iota`.

Reply:
> I wonder if we should directly pass SessionID here in well instead of relying on the context session? Not a requirement but if you think it is safer you can change it.

I am not 100% sure how we would do it. I _think_ it is the following:

1. Client-side calls `c.ResolveImageConfig()` (and sets the `ResolveImageConfigOpt{}` [here](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR853) and [here](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR849), in [`contextByName()`](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR817)
2. `contextByName` is called only by [`contextByNameFunc`](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR791)
3. `contextByNameFunc` is called only [here](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR459), passing it `client.Client` and `ocispecs.Platform`. 
4. As far as I can tell, about 20 lines above, it [derives the `SessionID`](https://github.com/moby/buildkit/pull/2827/files#diff-c876ea8bd639239ba60cde8cc40c2c2dd761f361564d4465b139b7b628e815bbR437)

If that is the same and correct `SessionID`, we can add it to the call chain. 

Is that it?

Reply:
OK, done. I am pretty sure I got it all, but if not, CI is our friend and will catch it.